src/common/socket.cpp

```diff
@@ -1635,12 +1635,34 @@ bool session_isActive(int32 fd)
 	return ( session_isValid(fd) && !session[fd]->flag.eof );
 }
 
+#ifdef HAVE_GETADDRINFO
+uint32_t host2ip(const char* hostname)
+{
+    struct addrinfo hints, *res;
```

Comment:
```suggestion
    addrinfo hints = { 0 }, *res;
```

Reply:
I think you mean ` = {}`, but i think memset is more correct here. I don't think we can assume addrinfo is default constructable, so we should continue using memset. 
Also using `struct` here for the system structs is nice, in case a system has a function named `addrinfo`.